core/src/main/java/com/linecorp/armeria/server/ServerBuilder.java

```diff
@@ -1708,12 +1709,26 @@ public ServerBuilder setHeaders(
     }
 
     /**
-     * Sets the {@link Supplier} which generates a {@link RequestId}.
+     * Sets the {@link Function} from {@link Supplier} which generates a {@link RequestId}.
      * By default, a {@link RequestId} is generated from a random 64-bit integer.
      *
      * @see RequestContext#id()
      */
     public ServerBuilder requestIdGenerator(Supplier<? extends RequestId> requestIdGenerator) {
+        final Supplier<? extends RequestId> requestIdSupplier = requireNonNull(requestIdGenerator);
+        this.requestIdGenerator = routingContext -> requestIdSupplier.get();
+        return this;
+    }
+
+    /**
+     * Sets the {@link Function} which generates a {@link RequestId}.
+     * Add logic to generate {@link RequestId} from {@link RoutingContext}.
```

Comment:
I modified it to **Generate {@link RequestId} from {@link RoutingContext}.** 
Will it be okay?

Reply:
I think I prefer the version I commented since it is more consistent with the javadoc for the overloaded method.

https://github.com/line/armeria/blob/bc48105148d652e2e48e788c94bec007882c31dc/core/src/main/java/com/linecorp/armeria/server/ServerBuilder.java#L1711